src/tribler/core/components/libtorrent/restapi/torrentinfo_endpoint.py

```diff
@@ -148,3 +152,19 @@ async def get_torrent_info(self, request):
             encoded_metainfo, ignore_errors=True), ensure_ascii=False).encode('utf-8'))
         return RESTResponse({"metainfo": encoded_metainfo,
                              "download_exists": download and not download_is_metainfo_request})
+
+    def get_infohash_from_magnet(self, magnet: Union[str, bytes]) -> Optional[bytes]:
+        """ Get the infohash from a magnet link.
+        Args:
+            magnet: A string representing the magnet link.
+        Returns:
+            bytes: The infohash of the magnet link. If the magnet link is invalid, None is returned.
+        """
+        infohash = None
+        try:
+            _, infohash, _ = parse_magnetlink(magnet)
+        except RuntimeError as e:
+            self._logger.warning(f'Error while getting an ingo hash from magnet: {e.__class__.__name__}: {e}',
+                                 exc_info=e)
+            self.last_error = e
```

Comment:
Since it's possible to have multiple simultaneous requests, I don't think it's a good idea to store exceptions like this. Sure, it will work right now, but if someone ever adds an await directly after calling this function, it could break.

Reply:
I can rewrite the whole endpoint in DownloadsEndpoint style:

```python
    async def add_download(self, request):
        ...
        try:
            download = await self.download_manager.start_download_from_uri(uri, config=download_config)
        except Exception as e:
            return RESTResponse({"error": str(e)}, status=HTTP_INTERNAL_SERVER_ERROR)

        return RESTResponse({"started": True, "infohash": hexlify(download.get_def().get_infohash())})

    async def start_download_from_uri(self, uri, config=None):
        """all the logic located here"""
```

Reply:
Rewriting the whole thing seems a bit much. You could just change `get_torrent_info` a bit so that so only need to parse the magnetlink once. Then drop `get_infohash_from_magnet`  and catch the exception DownloadsEndpoint-style

Reply:
Or, I could simply surround the `parse_magnetlink` call with a try-except block and leave the refactoring for later.

Reply:
Sure, that's fine too.

Reply:
I've implemented the second option, surrounding the `parse_magnetlink` call with a try-except block.